File: src/main/java/me/andre111/d20server/scripting/variable/PropertyVariableMap.java
package me.andre111.d20server.scripting.variable;

import me.andre111.d20common.model.Entity;
import me.andre111.d20server.scripting.Context;
import me.andre111.d20server.scripting.ScriptException;

public class PropertyVariableMap extends PropertyVariable {
	public PropertyVariableMap(String fullName, String propertyName) {
		super(fullName, propertyName);
	}

	@Override
	protected Entity getEntity(Context context) throws ScriptException {
		return context.map();
	}
}

File: core/common/scripting/variable/property-variable-map.js
import { PropertyVariable } from "./propertyvariable.js";

export class PropertyVariableMap extends PropertyVariable {
    constructor(fullName, propertyName) {
        super(fullName, propertyName);
    }

    getEntity(context) {
        if(!context.map) throw new Error('No map in this context');
        return context.map;
    }
}
